meson.build

```diff
@@ -0,0 +1,248 @@
+	'-Wrestrict',
+	'-Wreturn-local-addr',
+	'-fsanitize=cfi',
+	'-Wstringop-overflow',
+]
+
+possible_link_flags = [
+	'-Wl,--as-needed',
+	'-Wl,--gc-sections',
+	'-Wl,-z,relro',
+	'-Wl,-z,now',
+	'-pie',
+	'-Wl,-fuse-ld=gold',
+]
+
+if meson.version().version_compare('>=0.46')
+	add_project_link_arguments(cc.get_supported_link_arguments(possible_link_flags), language : 'c')
+else
+	add_project_link_arguments(possible_link_flags, language : 'c')
+endif
+
+add_project_arguments(cc.get_supported_arguments(possible_cc_flags), language : 'c')
+
+foreach ident : [
+        ['strlcpy',		'''#include <string.h>'''],
+        ['strlcat',		'''#include <string.h>'''],
+        ['pidfd_send_signal',	'''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['pidfd_open',		'''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+]
+
+        have = cc.has_function(ident[0], prefix : ident[1], args : '-D_GNU_SOURCE')
+        conf.set10('HAVE_' + ident[0].to_upper(), have)
+endforeach
+
+conf.set('_FILE_OFFSET_BITS', 64)
+
+libfuse = cc.find_library('fuse3', required : false)
+if libfuse.found()
+	conf.set10('HAVE_FUSE3', libfuse.found())
+	conf.set('FUSE_USE_VERSION', 30)
+else
+	libfuse = cc.find_library('fuse')
+	conf.set10('HAVE_FUSE', libfuse.found())
+	conf.set('FUSE_USE_VERSION', 26)
+endif
+
+libdl = cc.find_library('dl')
+threads = dependency('threads')
+
+config_h = configure_file(
+        output : 'config.h',
+        configuration : conf)
+config_include = include_directories('.')
+
+add_project_arguments('-include', 'config.h', language : 'c')
+
+subdir('tests')
+
+public_programs = []
+
+lxcfs_sources = files('src/lxcfs.c')
+
+public_programs += executable(
+        'lxcfs',
+        lxcfs_sources,
+        dependencies : [threads,
+			libdl,
+			libfuse],
+        install : true,
+        install_dir : bindir)
+
+liblxcfs_sources = files(
+	'src/api_extensions.h',
+	'src/bindings.c',
+	'src/bindings.h',
+	'src/cgroups/cgfsng.c',
+	'src/cgroups/cgroup.c',
+	'src/cgroups/cgroup.h',
+	'src/cgroups/cgroup2_devices.c',
+	'src/cgroups/cgroup2_devices.h',
+	'src/cgroups/cgroup_utils.c',
+	'src/cgroups/cgroup_utils.h',
+	'src/cgroup_fuse.c',
+	'src/cgroup_fuse.h',
+	'src/cpuset_parse.c',
+	'src/cpuset_parse.h',
+	'src/lxcfs.c',
+	'src/lxcfs_fuse_compat.h',
+	'src/macro.h',
+	'src/memory_utils.h',
+	'src/proc_cpuview.c',
+	'src/proc_cpuview.h',
+	'src/proc_fuse.c',
+	'src/proc_fuse.h',
+	'src/proc_loadavg.c',
+	'src/proc_loadavg.h',
+	'src/syscall_numbers.h',
+	'src/sysfs_fuse.c',
+	'src/sysfs_fuse.h',
+	'src/utils.c',
+	'src/utils.h')
+
+liblxcfs = shared_library(
+        'lxcfs',
+        liblxcfs_sources,
+        link_args : ['-shared'],
+        dependencies : [threads,
```

Comment:
-shared is already handled by the shared_library() function. Given the dependencies and liblxcfs_sources are shared between two targets, you could use:

```
commondep = declare_dependency(
    sources: liblxcfs_sources,
    dependencies: [threads, libdl, libfuse])
```

and then just have both shared_library() add a dependencies: commondep

Reply:
Thanks, done!